Document how to contribute a gallery example. (#4857)
scikit-image/scikit-image: CONTRIBUTING.txt

* All code should have tests (see `test coverage`_ below for more details).
* All code should be documented, to the same
  `standard <https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard>`_ as NumPy and SciPy.
* For new functionality, always add an example to the gallery (see

>   sphinx_gallery_ below for more details).

Reviewer: ```suggestion
  :ref:`_sphinx_gallery` below for more details).
```
?

Author: It should be
```rst
:ref:`sphinx_gallery`
```
then. I thought I would try the 'first method' (according to https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#explicit-links) but I'm happy to use `:ref:` everywhere (might be more standard, reading https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html).

Author: Done 69e41b51.

>   :ref:`sphinx_gallery` below for more details).

Reviewer 2: Isn't it
```suggestion
  :ref:`_sphinx_gallery` below for more details).
```
?

Reviewer 3: the updated link works, resolving this convo

Author: That's also what @jni was suggesting :thinking: but... see https://github.com/scikit-image/scikit-image/pull/4857#discussion_r461400636 and https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#use-a-custom-anchor

    make clean
    make html

> Gallery

Reviewer: I think this should go under the requirements *and* I just saw that the requirements are missing sphinx-gallery!

Author: Makes sense. Moving this [sub * n]section as a [sub * (n + 1)]subsection under the Requirements [sub * n]section with bdfbbd245.

~~~~~~~

If you are contributing an example to the

> `gallery <https://scikit-image.org/docs/dev/auto_examples/>`_ (or editing an

Author: Should this be an internal link (ref?) instead? There is another link to a https://scikit-image.org/ page elsewhere (line 83), so I left it this way...

Reviewer: I think a ref is better, good suggestion! Anyone else @scikit-image/core got an opinion?

Author: Done as part of bdfbbd24.

Then I thought I might as well change the other https://scikit-image.org/ link: 0fd5eb9.

Also, from https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html, it seems that the link text at
https://github.com/scikit-image/scikit-image/blob/7e4840bd9439d1dfb6beaf549998452c99f97fdd/CONTRIBUTING.txt#L256
is unnecessary, 'Coordinate conventions' being the title following the `numpy-images-coordinate-conventions` label (file `doc/source/user_guide/numpy_images.rst`).

existing one), write in
`reStructuredText <https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_.
Indeed, the gallery is a

> `Sphinx-Gallery <https://sphinx-gallery.github.io/stable/index.html>`_ one.

Author: "one" might be unnecessary

Reviewer: Agreed! 😂

Reviewer: The gallery is created using `Sphinx-Gallery <https://sphinx-gallery.github.io>`_. Refer to that page for complete instructions on syntax and usage.

Author: Thanks for this better wording! Included in bdfbbd2. I don't love my first sentence in the new [sub * (n + 1)]section ('The above-mentioned command includes the installation of Sphinx-Gallery.') but I thought repeating
```sh
pip install -r requirements/docs.txt
```
wouldn't be great either...

> Refer to the above to build (all) the docs, and check how your edits render at

Reviewer 4: Maybe mention the browser here, otherwise newcomers might not know how to check.

Author: Good point, realized that when replying: https://github.com/scikit-image/scikit-image/pull/4851#issuecomment-663026833
Done bdfbbd245.

``scikit-image/doc/build/html/auto_examples/`` (navigate to the file you have
added or changed).

> When adding an example, visit also

Reviewer 4: also visit

Author: I find there is a slight difference in meanings, but I'm not sure: I wrote 'visit also `index.html`' because the contributor was just told to *visit* another page (visit A, visit also B); 'also visit' sounds like the previous instruction would entail an action different from visiting a web page (e.g., run make html, also visit the resulting web pages).

``scikit-image/doc/build/html/auto_examples/index.html`` to check how the new
thumbnail renders on the gallery's homepage. To change the thumbnail image,

> please refer to this

Reviewer 4: Include both "this section" in the link

Author: Agreed. Done bdfbbd24.

    pip install -r requirements/docs.txt

.. _sphinx_gallery:

**Sphinx-Gallery:**

> The above-mentioned command includes the installation of Sphinx-Gallery.
> The gallery of :ref:`examples_gallery`
> is created using `Sphinx-Gallery <https://sphinx-gallery.github.io>`_.

Reviewer: ```suggestion
The above install command includes the installation of
`Sphinx-Gallery <https://sphinx-gallery.github.io>`_, which we use to create
the :ref:`examples_gallery`.
```

Author: Since
https://github.com/scikit-image/scikit-image/blob/c82a3c8c9d6187a056a43c7b7ccae26f6111cf93/doc/examples/README.txt#L1-L4
the sentence will read as "... to create the General examples" -- I'm fine with it.

> Refer to that page for complete instructions on syntax and usage.

Reviewer: ```suggestion
Refer to the Sphinx-Gallery documentation for complete instructions on syntax and usage.
```
